src/sympc/protocol/aby3/aby3.py

```diff
@@ -204,3 +206,86 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
         r_mpc = MPCTensor(shares=r, session=session, shape=x.shape)
         rPrime_mpc = MPCTensor(shares=rPrime, session=session, shape=x.shape)
         return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def local_decomposition(
+        x: ReplicatedSharedTensor, ring_size: str
+    ) -> List[ReplicatedSharedTensor]:
+        """Performs local decomposition to generate shares of shares.
+
+        Args:
+            x (ReplicatedSharedTensor) : input RSTensor.
+            ring_size (str) : Ring size to generate decomposed shares in.
+
+        Returns:
+            List[ReplicatedSharedTensor]: Decomposed shares in the given ring size.
+
+        Raises:
+            ValueError: If RSTensor does not have session uuid.
+        """
+        if x.session_uuid is None:
+            raise ValueError("Input RSTensor should have session_uuid")
+
+        session = get_session(x.session_uuid)
+        ring_size = int(ring_size)
+        tensor_type = get_type_from_ring(ring_size)
+        rank = session.rank
+        nr_parties = session.nr_parties
```

Comment:
Q: Maybe we can "hardcode" the number of parties to 3 for this? (since ABY3 is 3 parties)
maybe have a global `Final` variable named `NR_PARTIES = 3`?
(Maybe we can also add a check method to validate that we have the correct number of parties before calling any method?)

Reply:
ABY3 has this mix of protocols george where some are three parties, and some are N parties(`truncation 2 algoritm`),I would have this sanity check for protocols which support exactly three parties :+1: